### PR TITLE
Allow use of validation loss for early stopping

### PR DESCRIPTION
**Pull request by turian**

### CODE REVIEW

File: heareval/predictions/task_predictions.py [hunk 1]
```diff
@@ -203,10 +203,12 @@ def __init__(
         prediction_type: str,
         scores: List[ScoreFunction],
         conf: Dict,
+        use_scoring_for_early_stopping: bool = True,
```
Comment (turian) [inline heareval/predictions/task_predictions.py:206]: https://github.com/neuralaudio/hear-eval-kit/issues/347

File: heareval/predictions/task_predictions.py [hunk 2]
```diff
@@ -431,79 +453,86 @@ def _score_epoch_end(self, name: str, outputs: List[Dict[str, List[Any]]]):
...
             # be stored as the "best" epoch
             epoch = self.current_epoch + 1
             postprocessing_cached = None
         elif name == "test":
             epoch = self.current_epoch
-            postprocessing_cached = self.epoch_best_postprocessing[epoch]
+            postprocessing_cached = self.epoch_best_postprocessing_or_default(epoch)
         else:
             raise ValueError
         # print("\n\n\n", epoch)
```
Comment (turian) [inline heareval/predictions/task_predictions.py:474]: All these red lines get indented basically

File: heareval/predictions/task_predictions.py [hunk 3]
```diff
@@ -413,6 +424,17 @@ def __init__(
         self.epoch_best_postprocessing: Dict[int, Tuple[Tuple[str, Any], ...]] = {}
         self.postprocessing_grid = postprocessing_grid
 
+    def epoch_best_postprocessing_or_default(
+        self, epoch: int
+    ) -> Tuple[Tuple[str, Any], ...]:
+        if self.use_scoring_for_early_stopping:
+            return self.epoch_best_postprocessing[epoch]
+        else:
+            postprocessing_confs = list(ParameterGrid(self.postprocessing_grid))
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:433]: where do we define the postprocessing grids for secret tasks?

**Review by jorshi — APPROVED** [review]
This all looks good - ran on my machine with the updated task and it is working

File: heareval/predictions/task_predictions.py (hunk 3 above)
Comment (turian) [reply heareval/predictions/task_predictions.py:433]: It's the default unless we override it using config `evaluation_params: event_postprocessing_grid: {}`

**Pull request merged**